python/instrumentation/openinference-instrumentation-vertexai/src/openinference/instrumentation/vertexai/_wrapper.py

```diff
@@ -0,0 +1,469 @@
+import base64
+import inspect
+import json
+import logging
+from functools import singledispatch
+from itertools import chain
+from typing import (
+    Any,
+    AsyncIterable,
+    Awaitable,
+    Callable,
+    ContextManager,
+    Iterable,
+    Iterator,
+    Mapping,
+    Optional,
+    Protocol,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import proto
+import wrapt
+from google.cloud import aiplatform_v1 as v1  # Stable
+from google.cloud import aiplatform_v1beta1 as v1beta1  # Supports latest preview features
+from openinference.instrumentation import get_attributes_from_context, safe_json_dumps
+from openinference.instrumentation.vertexai import _status
+from openinference.instrumentation.vertexai._accumulator import (
+    _IndexedAccumulator,
+    _KeyValuesAccumulator,
+    _StringAccumulator,
+)
+from openinference.instrumentation.vertexai._proxy import _proxy
+from openinference.semconv.trace import (
+    ImageAttributes,
+    MessageAttributes,
+    MessageContentAttributes,
+    OpenInferenceMimeTypeValues,
+    OpenInferenceSpanKindValues,
+    SpanAttributes,
+    ToolCallAttributes,
+)
+from opentelemetry import context as context_api
+from opentelemetry import trace as trace_api
+from opentelemetry.context import _SUPPRESS_INSTRUMENTATION_KEY
+from opentelemetry.trace import Span, Status, StatusCode, use_span
+from opentelemetry.util.types import AttributeValue
+from typing_extensions import TypeAlias
+
+__all__ = ("_Wrapper",)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+T = TypeVar("T")
+
+# https://cloud.google.com/vertex-ai/docs/reference#versions
+# - v1: Stable
+# - v1beta1: Supports latest preview features
+Content: TypeAlias = Union[v1.Content, v1beta1.Content]
+Candidate: TypeAlias = Union[v1.Candidate, v1beta1.Candidate]
+Part: TypeAlias = Union[v1.Part, v1beta1.Part]
+UsageMetadata: TypeAlias = Union[
+    v1.GenerateContentResponse.UsageMetadata,
+    v1beta1.GenerateContentResponse.UsageMetadata,
+]
+GenerateContentRequest: TypeAlias = Union[
+    v1.GenerateContentRequest,
+    v1beta1.GenerateContentRequest,
+]
+GenerateContentResponse: TypeAlias = Union[
+    v1.GenerateContentResponse,
+    v1beta1.GenerateContentResponse,
+]
+
+_PREDICTION_SERVICE_REQUESTS = sorted(
+    {
+        name
+        for module in (
+            # https://cloud.google.com/vertex-ai/docs/reference#versions
+            # - v1: Stable
+            # - v1beta1: Supports latest preview features
+            v1.types.prediction_service,
+            v1beta1.types.prediction_service,
+        )
+        for name, _ in inspect.getmembers(
+            module,
+            lambda cls: inspect.isclass(cls)
+            and issubclass(cls, proto.Message)
+            and cls.__name__.endswith("Request"),
+        )
+    }
+)
+
+
+class _Wrapper:
+    _status = _status
```

Comment:
docstring to explain that this is the status of whether or not it's instrumented. Not clear otherwise.